packages/controllers/src/snaps/SnapController.ts

```diff
@@ -53,31 +60,75 @@ type RequestedSnapPermissions = {
   [permission: string]: Record<string, Json>;
 };
 
-export type SerializableSnap = {
-  initialPermissions: RequestedSnapPermissions;
+/**
+ * A Snap as it exists in {@link SnapController} state.
+ */
+export type Snap = {
+  /**
+   * Whether the Snap is enabled, which determines if it can be started.
+   */
+  enabled: boolean;
+
+  /**
+   * The ID of the Snap.
+   */
   id: SnapId;
-  permissionName: string;
-  version: string;
+
+  /**
+   * The initial permissions of the Snap, which will be requested when it is
+   * installed.
+   */
+  initialPermissions: RequestedSnapPermissions;
+
+  /**
+   * The Snap's manifest file.
+   */
   manifest: SnapManifest;
-  status: SnapStatus;
-  enabled: boolean;
-};
 
-export type TruncatedSnap = Pick<
-  SerializableSnap,
-  'id' | 'initialPermissions' | 'permissionName' | 'version'
->;
+  /**
+   * The name of the permission used to invoke the Snap.
+   */
+  permissionName: string;
 
-export type Snap = SerializableSnap & {
+  /**
+   * The source code of the Snap.
+   */
   sourceCode: string;
+
+  /**
+   * The current status of the Snap, e.g. whether it's running or stopped.
+   */
+  status: SnapStatus;
+
+  /**
+   * The SVG icon of the Snap.
+   */
+  svgIcon: string | null;
+
+  /**
+   * The version of the Snap.
+   */
+  version: string;
 };
 
+/**
+ * A {@link Snap} object with the fields that are relevant to an external
+ * caller.
+ */
+export type TruncatedSnap = Pick<Snap, TruncatedSnapFields>;
+
 export type SnapError = {
   message: string;
   code: number;
   data?: Json;
 };
 
+type FetchSnapResult = {
```

Comment:
Would `FetchedSnapResult` be a slightly better name?

Reply:
I don't think so, because this is referring to the return type of the `_fetchSnap` methods. I did add some docstrings for this type here though: 892e9fa

Reply:
Ahh okay, that's cool then